autogaita/autogaita_dlc_gui.py

```diff
@@ -1374,6 +1355,91 @@ def find_number(fullstring, prestring, poststring):
     return int(fullstring[start_idx:end_idx])
 
 
+def update_config_file(results, cfg):
+    """updates the dlc_gui_config file with this runs parameters"""
+    # transform tkVars into normal strings and bools
+    output_dicts = [{}, {}]
+    for i in range(len(output_dicts)):
+        if i == 0:
+            input_dict = results
+        elif i == 1:
+            input_dict = cfg
+        for key in input_dict.keys():
+            if key in LIST_VARS:
+                # if list of strings, initialise output empty list and get & append vals
+                output_dicts[i][key] = []
+                for list_idx in range(len(input_dict[key])):
+                    output_dicts[i][key].append(input_dict[key][list_idx].get())
+            elif key in DICT_VARS:
+                # if dict of list of strings, initialise as empty dict and assign stuff
+                # key = "angles" or other DICT_VAR
+                # inner_key = "name" & "lower_ / upper_joint"
+                # list_idx = idx of list of strings of inner_key
+                output_dicts[i][key] = {}
+                for inner_key in input_dict[key]:
+                    output_dicts[i][key][inner_key] = []
+                    for list_idx in range(len(input_dict[key][inner_key])):
+                        output_dicts[i][key][inner_key].append(
+                            input_dict[key][inner_key][list_idx].get()
+                        )
+            else:
+                output_dicts[i][key] = input_dict[key].get()
+
+    # merge the two configuration dictionaries
+    configs_list = [output_dicts[0], output_dicts[1]]  # 0 = results, 1 = cfg, see above
+    # write the configuration file
+    with open(
+        os.path.join(AUTOGAITA_FOLDER_PATH, CONFIG_FILE_NAME), "w"
+    ) as config_json_file:
+        json.dump(configs_list, config_json_file, indent=4)
+
+
+def extract_cfg_from_json_file(root):
+    """loads the cfg dictionary from the config file"""
+    # load the configuration file
+    with open(
+        os.path.join(AUTOGAITA_FOLDER_PATH, CONFIG_FILE_NAME), "r"
+    ) as config_json_file:
+        # config_json contains list with 0 -> result and 1 -> cfg data
+        last_runs_cfg = json.load(config_json_file)[1]
+
+    cfg = {}
+    # assign values to the cfg dict
+    for key in last_runs_cfg.keys():
+        if key in TK_BOOL_VARS:
+            cfg[key] = tk.BooleanVar(root, last_runs_cfg[key])
+        elif key in LIST_VARS:
+            cfg[key] = []
+            for entry in range(len(last_runs_cfg[key])):
```

Comment:
entry is not the value but the index (because of for range(len()) syntax) - is this wanted?